ndb/tests/conftest.py

```diff
@@ -36,3 +36,19 @@ def property_clean_cache():
     finally:
         assert model.Property._FIND_METHODS_CACHE != {}
         model.Property._FIND_METHODS_CACHE.clear()
+
+
+@pytest.fixture
+def reset_kind_map():
+    """Reset ``Model._kind_map``.
+
+    This mapping of "kind" to class is set whenever a new subclass of ``Model``
+    is created. We create ``Model`` subclasses in tests and don't want those
+    definitions to leak to other tests, so this fixture resets the mapping to
+    its value before the text. (Since some classes might be declared at module
+    scope, we can't just clear the mapping altogether.)
+    """
+    previous = model.Model._kind_map
```

Comment:
ISTM it may be easier to reason about if we keep `Model._kind_map == {}` as a requirement. This will help us catch tests which are polluting the hash. So the global `ManyFields` will probably have to move to a helper function or method which returns that type

Or is there a way to add a hook in `pytest` for **every** test (i.e. so we don't have to explicitly use the fixture).

Reply:
Yeah, we can certainly move ManyFields or use a factory function.

It is also possible to declare `reset_kind_map` with `autouse=True`, and it will get used by every test in our suite. While that might seem a little indiscriminate, it's a very cheap operation computationally, so I don't really have a problem with that approach.

Reply:
If we do it for `reset_kind_map` I propose we also do it for `property_clear_cache` or consolidate them to an omnibus `reset_all_the_things` fixture.

Reply:
:+1: On `autouse=True` and on the omnibus.